Significant update check description

SurajThotakura opened the pull request:
1. Updated icons with SVG instead of using icomoon. (Tried to use css.gg but the icons were very limited).

2. Added favicon to the site.

3. Added typing test functionality
    a. Used a textarea with a "Keyup" event listner.
    b. This method causes a UX issue where the typed content doesn't align properly when there is a line break.
    c. Has a bug to restart the test due to the event listener not terminating.

4. Smaller modifications based on previous review.

SurajThotakura committed:
Significant update check description

1. Updated icons with SVG instead of using icomoon. (Tried to use css.gg but the icons were very limited).

2. Added favicon to the site.

3. Added typing test functionality
    a. Used a textarea with a "Keyup" event listner.
    b. This method causes a UX issue where the typed content doesn't align properly when there is a line break.
   c. Has a bug to restart the test due to the event listener not terminating.

4. Smaller modifications based on previous review.

akhil0001 commented on package.json:
```diff
@@ -18,6 +18,12 @@
+    "json-data-extender": "^0.7.1",
+    "parcel": "^2.0.0-beta.3.1",
+    "txtgen": "^2.2.8",
+    "typescript": "^4.3.2"
```
doubt: Why Typescript?

akhil0001 commented on src/index.html:
```diff
@@ -5,21 +5,46 @@
+    <link rel="icon" type="image/png" sizes="192x192"  href="./favicon/android-icon-192x192.png">
+    <link rel="icon" type="image/png" sizes="32x32" href="./favicon/favicon-32x32.png">
+    <link rel="icon" type="image/png" sizes="96x96" href="./favicon/favicon-96x96.png">
+    <link rel="icon" type="image/png" sizes="16x16" href="./favicon/favicon-16x16.png">
```
doubt: Why do we need these many favicons ?

SurajThotakura replied:
To address all the variation of the fav icons used in different browsers, operating systems and places, multiple dimensions and formats of favicons are uploaded.

akhil0001 commented on src/index.js:
```diff
@@ -24,17 +36,180 @@ function openTab(i) {
+
+function countDown(){
+    console.log('countdown started');
+    let haha = setInterval(() => {
```
fix: This needs to be changed. haha is not good variable :)

akhil0001 commented on src/index.js:
```diff
@@ -24,17 +36,180 @@ function openTab(i) {
+    openTab(0);
+}
+
+startButton.addEventListener('click', () => {
```
fix: Let's prefer a named function for event listeners. Major Reasons:
1. Code reusability
2. Easy way to remove the event listener
3. Readable code

akhil0001 commented on src/index.js:
```diff
@@ -24,17 +36,180 @@ function openTab(i) {
+    countDown();
+
+    // validation
+    typeUpdate.addEventListener('keyup', (e) => {
```
fix: As mentioned above, lets use named functions instead of anonymous functions

akhil0001 commented on src/index.js:
```diff
@@ -24,17 +36,180 @@ function openTab(i) {
+    let b = e.key;
+    console.log(a, b, i, "errors = " + j, k, w);
+    k++;
+    if(a==b){
```
fix: Prefer `===` over `==` . == is not recommended in javascript since it comes with some pains of type coercion
Ref: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Equality_comparisons_and_sameness

akhil0001 commented on src/index.js:
```diff
@@ -24,17 +36,180 @@ function openTab(i) {
+        };
+        i++;
+    }
+    else if(b == 'Backspace'){
```
suggestion: We could refactor these conditions into simple object-based code. 
Ref: https://medium.com/swlh/stop-using-if-else-statements-f4d2323e6e4

akhil0001 replied:
@akhil0001 Create a fork from this branch to refactor the code

akhil0001 commented on src/styles.css:
```diff
@@ -4,11 +4,22 @@
 }
 
+:root{
+    --blue: #e3f2ff;
```
praise: nice usage of css variables 👍

akhil0001 commented on src/styles.css:
```diff
@@ -90,34 +106,86 @@ body > footer {
+    height: 7.9rem;
+    columns: auto;
+    overflow: hidden;
+    border-radius: 2px;
```
doubt: I am not sure if this works when the border is none.

akhil0001 commented on public/favicon/manifest.json:
```diff
@@ -0,0 +1,41 @@
+{
+ "name": "App",
```
fix: Lets not include public/ into reviews. So we have to add this to `.gitignore`

akhil0001 reviewed (changes requested):
I think we could use some code refactoring to make it clean. Lets have this as agenda for upcoming meeting

akhil0001 committed:
refactor: improve handleValidaition function

SurajThotakura committed:
minor fixes to make the validator work

While accessing variables from global their values were not being updated. Fixed it by calling them directly as global.variable ...

SurajThotakura committed:
Solved the object scope issue

re populated the changed variables at the end of the event-listener.
